### DISCUSSION TITLE
[#1005963] Grapple doesn't work with Selection Modes other than Freeform

### DISCUSSION THREAD

**Discussion by CaptainCooky** [Mojang/minecraft-editor#89] [Feedback]
The Grapple ability doesn't work in Fixed-Distance and Adjacent Selection modes. 

This may be by design. That said, the Grapple is a highly useful function and having to swap into Freeform Select Mode to use it seems counter-intuitive. 

Does anyone else have thoughts on this?

**Comment by minecraftchloe** [comment]
I could see it being helpful in Adjacent! Fixed Distance _maybe?_ If you scroll the wheel out far enough where you get some distance to actually grapple to?

**Reply by CaptainCooky** [reply]
Agreed, Fixed-distance might be a bit awkward! 
I wonder if you could implement a max distance hotkey for Fixed-distance mode. This might serve 2 purposes as we then have the ability to easily select a block at the Max possible distance from us, and as a secondary use, it could work in conjunction with the grapple. 

Too complicated??😁

**Reply by dacowan** [reply]
This is an oversight on my part - I tied grapple to the 3d block cursor -- it should be just a raycast from the mouse to the world, irrespective of cursor mode.  I'll add it to my "not-inconsiderable" list of things to do :)
